### PR TITLE
versatile-data-kit: Establish release process

### PR DESCRIPTION
**Pull request by gageorgiev**
This change describes the release process for publicly-declared
releases of VDK.

Signed-off-by: Gabriel Georgiev <gageorgiev@vmware.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -95,3 +95,13 @@ Entrypoint of CICD is the file .gitlab-ci.yml.
 
 There you can find the full definition of the CI/CD pipeline.
 For more details see the [CICD wiki in our Dev Guide](https://github.com/vmware/versatile-data-kit/wiki/Gitlab-CICD)
+
+# How to make a new VDK release
+
+To make a new stable release, follow these steps:
+- Create a new release from the Releases menu - this will require a new tag;
+- Autogenerate the changelog, then sort the changes alphabetically; if all change names are prefixed with the component name, this will ensure changes are grouped by component. Sorting can be done using the `sort` utility available in MacOS and Linux distros;
```
Comment (doks5) [inline CONTRIBUTING.md:103]: Maybe add a link with steps how to auto-generate the changelog?

**Review by doks5 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (gageorgiev) [reply CONTRIBUTING.md:103]: noted, will link to the github docs to make it clearer

**Review by tozka — APPROVED** [review]
Let's try the whole process (except for the tweet part) . And see how it looks from user point of view and how easy it is to follow. And we can revise if necessary.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -95,3 +95,13 @@ Entrypoint of CICD is the file .gitlab-ci.yml.
...
+- Review which components have received changes since the previous release - this should be apparent from the changelog;
+- Create a new PR which bumps the minor versions of all newly changed components - plugins' version source of truth is located inside their respective `setup.py` files;
```
Comment (tozka) [inline CONTRIBUTING.md:105]: be more specific or maybe add link to "release" section for vdk-core , control-service

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -95,3 +95,13 @@ Entrypoint of CICD is the file .gitlab-ci.yml.
 
 There you can find the full definition of the CI/CD pipeline.
 For more details see the [CICD wiki in our Dev Guide](https://github.com/vmware/versatile-data-kit/wiki/Gitlab-CICD)
+
+# How to make a new VDK release
+
+To make a new stable release, follow these steps:
```
Comment (tozka) [inline CONTRIBUTING.md:101]: let's call it public release (minor or major version release). All releases should be stable as long as all tests have passed.

**Review by tozka — COMMENTED** [review]
Let's try the whole process (except for the tweet part) . And see how it looks from user point of view and how easy it is to follow. And we can revise if necessary.

File: CONTRIBUTING.md (hunk 2 above)
Comment (gageorgiev) [reply CONTRIBUTING.md:105]: noted

File: CONTRIBUTING.md (hunk 3 above)
Comment (gageorgiev) [reply CONTRIBUTING.md:101]: noted

**Review by doks5 — APPROVED** [review]
LGTM

**Pull request merged**